include/secp256k1_musig.h

```diff
@@ -375,26 +171,207 @@ SECP256K1_API int secp256k1_musig_partial_signature_serialize(
  *  encoded numbers are out of range, signature verification with it is
  *  guaranteed to fail for every message and public key.
  */
-SECP256K1_API int secp256k1_musig_partial_signature_parse(
+SECP256K1_API int secp256k1_musig_partial_sig_parse(
     const secp256k1_context* ctx,
-    secp256k1_musig_partial_signature* sig,
+    secp256k1_musig_partial_sig* sig,
     const unsigned char *in32
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
+/** Computes a aggregate public key and the hash of the given public keys.
+ *
+ *  Different orders of `pubkeys` result in different `agg_pk`s.
+ *
+ *  The pubkeys can be sorted before combining with `secp256k1_xonly_sort` which
+ *  ensures the same resulting `agg_pk` for the same multiset of pubkeys.
+ *  This is useful to do before pubkey_agg, such that the order of pubkeys
+ *  does not affect the aggregate public key.
+ *
+ *  Returns: 1 if the public keys were successfully aggregated, 0 otherwise
+ *  Args:        ctx: pointer to a context object initialized for verification
+ *           scratch: scratch space used to compute the aggregate pubkey by
+ *                    multiexponentiation. If NULL, an inefficient algorithm is used.
```

Comment:
What's the right way to know how much space should be allocated for this?
This is a general question about the current scratch/batching API